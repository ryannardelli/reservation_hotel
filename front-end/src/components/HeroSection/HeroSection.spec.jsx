import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { HeroSection } from '.';

test('should render with title and paragraph in screen', () => {
  render(<HeroSection />);
  const title = screen.getByText(/hotel sol e mar/i);
  const subtitle = screen.getByText(/faça já a sua reserva!/i);
  console.log(screen.debug);
  expect(title).toBeInTheDocument();
  expect(subtitle).toBeInTheDocument();
});

test('shound render a h1 and a paragraph', () => {
  render(<HeroSection />);
  const h1Element = screen.getByRole('heading', { level: 1 });
  const paragraphElement = screen.getByRole('paragraph');
  expect(h1Element).toBeInTheDocument();
  expect(paragraphElement).toBeInTheDocument();
});

test('matches snapshot', () => {
  const { asFragment } = render(<HeroSection />);
  expect(asFragment()).toMatchSnapshot();
});
