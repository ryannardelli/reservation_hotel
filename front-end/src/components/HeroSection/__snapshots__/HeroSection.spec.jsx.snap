// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Hotel Sol e Mar
    </h1>
    <p>
      Faça já a sua reserva!
    </p>
  </div>
</DocumentFragment>
`;
